pkg/advisory/discover_aliases.go

```diff
@@ -0,0 +1,150 @@
+package advisory
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"sort"
+
+	"github.com/samber/lo"
+	"github.com/wolfi-dev/wolfictl/pkg/configs"
+	v2 "github.com/wolfi-dev/wolfictl/pkg/configs/advisory/v2"
+	"github.com/wolfi-dev/wolfictl/pkg/vuln"
+)
+
+// DiscoverAliasesOptions is the set of options for the DiscoverAliases
+// function.
+type DiscoverAliasesOptions struct {
+	// AdvisoryDocs is the Index of advisory documents on which to operate.
+	AdvisoryDocs *configs.Index[v2.Document]
+
+	// AliasFinder is the alias finder to use for discovering aliases for the given
+	// vulnerabilities.
+	AliasFinder AliasFinder
+
+	// SelectedPackages is the list of packages to operate on. If empty, all
+	// packages will be operated on.
+	SelectedPackages []string
+}
+
+// DiscoverAliases queries external data sources for aliases for the
+// vulnerabilities described in the selected advisories and updates the advisory
+// documents with the discovered aliases.
+func DiscoverAliases(ctx context.Context, opts DiscoverAliasesOptions) error {
+	documents := opts.AdvisoryDocs.Select().Configurations()
+
+	// If the caller selected specific packages (by name), filter the set of
+	// advisory documents to just those packages.
+	if len(opts.SelectedPackages) > 0 {
+		documents = lo.Filter(documents, func(doc v2.Document, _ int) bool {
+			return slices.Contains(opts.SelectedPackages, doc.Package.Name)
+		})
+	}
+
+	for _, doc := range documents {
+		for _, adv := range doc.Advisories {
+			// If the advisory ID is already a CVE ID, do a lookup and ensure the discovered
+			// aliases are present in the advisory's list of aliases.
+			if vuln.RegexCVE.MatchString(adv.ID) {
+				ghsas, err := opts.AliasFinder.GHSAsForCVE(ctx, adv.ID)
+				if err != nil {
+					return err
+				}
```

Comment:
If I'm reading this correctly, we'll send a GitHub request for each CVE. This may be premature optimization, but in some cases we have a CVE showing up in many packages. In that case, we'd end up requesting the same information from GitHub repeatedly.

Reply:
That's right. I had mentioned this in the PR description as a potential enhancement:

>6. Cache API queries in memory during command execution, since there's a decent amount of redundancy today of vulnerability IDs in our advisory data set.

Would following up with this optimization be okay with you? Or would you rather see it get in immediately?

Reply:
Ah - all good. Let's keep that as a future improvement